### DISCUSSION TITLE
/using-pihole-with-pfsense-or-opnsense/

### DISCUSSION THREAD

**Discussion by migratecomments** [ironicbadger/blog.ktz.me#8] [Announcements]
Using Pihole with pfsense or opnsense
        
[https://blog.ktz.me/using-pihole-with-pfsense-or-opnsense/](https://blog.ktz.me/using-pihole-with-pfsense-or-opnsense/)

**Comment by migratecomments** [comment]
```header
Originally posted by deliarmin on 2020-11-27 14:52
```

Hi Alex!  
Thx for the post! I also want to add PiHole to my OPNsense setup, but I have one question: Is there a specific reason, why you put the DNS servers in this order? Why not adding Pihole as a DNS Server to OPNsense and let the clients query OPNsense?  
Thx!  
Armin

**Reply by migratecomments** [reply]
```header
Originally posted on 2021-03-26 15:35
```

I found this approach leaky. Not all ads were blocked this way every time.

**Comment by migratecomments** [comment]
```header
Originally posted by gvfarns on 2021-07-22 02:24
```

Using this method, you need to identify each client individually and instruct OPNsense to tell it to use pihole? Wouldn't it work to just specify pihole as the only global DNS on opnsense (system|general|networking|dns servers). Then create a single DHCP static mapping for the pihole in which you give it 8.8.8.8. If your network has more than just a few computers, your proposed solution would take quite a while to configure, no? I think there is something here I don't understand fully.

**Comment by migratecomments** [comment]
```header
Originally posted by billy wobbledagger on 2022-07-31 03:34
```

Hi Alex,

Since I found your blogs and PMS site I've built a really good Proxmox server using all the things you've shown. It's an old Ugel H820c thin client, 8GB ram and Celeron 847 (2) @ 1.100GHz and a dual Intel gigabit lan card for ethernet bonding, that works great too. I've got an external HDD enclosure 12TB with Snapraid and Mergerfs (as learned from your site). It has the odd slow down but only when I hit it hard!

I've slowly been learning Docker, which I've come to really appreciate. I used Pihole and have finally managed to get a Docker version up and running. It is noticeably faster than my RPi3b that has now been demoted to the standby pihole, now in line for decommissioning.

I also love OPNSense but unfortunately being BSD based I can't get higher than 250Mbits as it uses PPPoE. My BT FTTP is 500/125 so I have to use either the BT router or an openwrt router in between and I'm actually trying to reduce boxes (and power useage!). On OPNsense using ethernet on the same port I get full gigabit throughput, but not on PPPoE. Hopefully one day the BSD's will fix this. Pihole is fantastic and my wife hates it as much as I love it. She now has her very own network, complete with full adverts - which she loves!  
Keep up the good work mate,  
Hopmeister

**Comment by alhaddar** [comment]
Trying to understand dig. When I run a dig command, the "SERVER:" section always comes back with 127.0.0.1#53

;; Query time: 3 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)

Sorry if this question is foolish in any way, I really dont understand how it works. Thanks
